service-tests/mozilla.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+const Joi = require('joi');
+const ServiceTester = require('./runner/service-tester');
+
+const t = new ServiceTester({ id: 'amo', title: 'Mozilla Addons' });
+module.exports = t;
+
+t.create('Downloads')
+  .get('/d/IndieGala-Helper.json')
+  .expectJSONTypes(Joi.object().keys({
+    name: Joi.equal('downloads'),
+    value: Joi.string().regex(/^\d+k$/)
+  }));
+
+t.create('Version')
+  .get('/v/IndieGala-Helper.json')
+  .expectJSONTypes(Joi.object().keys({
+    name: Joi.equal('mozilla add-on'),
+    value: Joi.string().regex(/^v\d+(\.\d+)?(\.\d+)?$/)
+  }));
+
+t.create('Version - Custom label')
+  .get('/v/IndieGala-Helper.json?label=IndieGala Helper')
+  .expectJSONTypes(Joi.object().keys({
+    name: Joi.equal('IndieGala Helper'),
+    value: Joi.string().regex(/^v\d+(\.\d+)?(\.\d+)?$/)
+  }));
+
+t.create('Users')
+  .get('/users/IndieGala-Helper.json')
+  .expectJSONTypes(Joi.object().keys({
+    name: Joi.equal('users'),
+    value: Joi.string().regex(/^\d+$/)
+  }));
+
+t.create('Rating')
+  .get('/rating/IndieGala-Helper.json')
+  .expectJSONTypes(Joi.object().keys({
+    name: Joi.equal('rating'),
+    value: Joi.string().regex(/^\d\/\d$/)
+  }));
+
+t.create('Stars')
+  .get('/stars/IndieGala-Helper.json')
+  .expectJSONTypes(Joi.object().keys({
+    name: Joi.equal('rating'),
+    value: Joi.string()
```

Comment:
not sure how to do this one correctly, anyone got any ideas?
this is the json for 4/5 stars:
```json
{
  "name": "rating",
  "value": "â˜…â˜…â˜…â˜…â˜†"
}
```

Reply:
I read [the JSON spec](https://tools.ietf.org/html/rfc7159#section-7) and a few other things, and it's clear that, while escaping JSON is allowed, it's also totally acceptable to return almost any Unicode character within a JSON string, and also not necessary to specify the content type in a JSON HTTP call. Firefox and the terminal display [the JSON result](https://img.shields.io/amo/stars/IndieGala-Helper.json) just fine:

```json
{
  "name": "rating",
  "value": "★★★★☆"
}
```

I'm seeing the same as you in Chrome and Safari, and don't completely understand why.

After some playing around, it looks like `Joi.string().regex(/^[\u2605\u2606]{5}$/)` will work.